### ISSUE TITLE
[Bug] first_admin_response_at can be grouped under "part_type= 'assignment'"

### ISSUE THREAD

**Issue by TrangNgo** [fivetran/dbt_intercom#35]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the issue

Hi team, I'm opening an issue on behalf of the customer. 

This is related to the issue posted here, where we are grabbing the 2nd admin comment as the "first admin response":
https://community.fivetran.com/t5/working-with-data/inconsistency-with-intercom-dbt-data/td-p/642

Intercom support has confirmed that the first admin comment can be grouped under part_type=assignment:
> On the issue of how the [Fivetran integration](https://app.intercom.com/a/apps/_/appstore?app_package_code=fivetran-6z0a) analyzes Intercom data from admin replies in conversations:
> 
> It looks like the issue is around an admin reply being coupled under "assignment" when they occur at the same time, is that correct? And therefore the conversation part type is "assignment", rather than "comment".
> 
> Looking at this conversation: https://app.intercom.com/a/apps/urub4jwk/conversations/7758
> 
> The admin Jesus's first reply is grouped with an "assignment" at 2023-01-26 17:10:48 UTC. Then 2nd reply from Jesus is at 2023-01-26 17:23:15 UTC and it is a part type "comment".
> 
> This is because, when [conversations are self assigned from an admin reply](https://app.intercom.com/a/apps/_/settings/assignment-preferences), the first reply will be grouped under "assignment"
![image](https://user-images.githubusercontent.com/25066283/223564574-1f421e19-2d5f-4d77-aaef-3a623716f0d4.png)
> In the [Conversation Model / Object](https://developers.intercom.com/intercom-api-reference/reference/the-conversation-model) it would look like this
![image](https://user-images.githubusercontent.com/25066283/223564638-2a0b1e64-f4c7-42b0-a8c9-fdd67e8abd79.png)
> Then subsequent admin replies would be a "comment"
![image](https://user-images.githubusercontent.com/25066283/223564703-00a326f1-17df-4b18-9a66-9204481f0cf1.png)
> So if Fivetran is ignoring a reply in the "body" section of the "assignment" part type, they would need to address this on their end, as this is the expected behavior in Intercom of how this data is grouped

### Expected behavior

For the first_admin_response_at, we also need to check if the first admin response is coming through as part_type=assignment.

### dbt Project configurations

n/a

### Package versions

n/a

### What database are you using dbt with?

snowflake

### dbt Version

n/a

### Additional Context

This is related to the issue posted here, where we are grabbing the 2nd admin comment as the "first admin response":
https://community.fivetran.com/t5/working-with-data/inconsistency-with-intercom-dbt-data/td-p/642

### Are you willing to open a PR to help address this issue?

- [ ] Yes.
- [ ] Yes, but I will need assistance and will schedule time during our [office hours](https://calendly.com/fivetran-solutions-team/fivetran-solutions-team-office-hours) for guidance
- [ ] No.

**Comment by fivetran-joemarkiewicz** [comment]
Thanks so much for opening this issue @TrangNgo and for helping us isolate the issue!

I believe I found the fix (with the help of your call out) and currently have an open PR that should address the original issue. Essentially, I adjusted the admin_response_at metrics to take into account assignment part_types if the body of the message is not null. This should resolve the initial error. If you wish, you can test the changes by using the version of the package in your packages.yml.

```yml
packages:
  - git: https://github.com/fivetran/dbt_intercom.git
    revision: bugfix/first-admin-response
    warn-unpinned: false 
```

**Comment by fivetran-joemarkiewicz** [comment]
Closing this issue as the above mentioned PRs have been merged and is live within the `v0.6.1` version of the package.
 
Please feel free to reopen if the issue persists on your end.
